submissions/andrewklmn/a-tiny-js-world/index.js

```diff
@@ -4,58 +4,86 @@
    Code repository: https://github.com/andrewklmn/a-tiny-JS-world
    Web app: https://andrewklmn.github.io/a-tiny-JS-world/
    */
-
 // ======== OBJECTS DEFINITIONS ========
-const dog = {
-  species: 'dog',
-  name: 'Dyuka',
-  gender: 'male',
-  legs: 4,
-  hands: 0,
-  saying: 'woof!',
-  friends: [],
-};
-
-const cat = {
-  species: 'cat',
-  name: 'Barsik',
-  gender: 'male',
-  legs: 4,
-  hands: 0,
-  saying: 'meow!',
-  friends: [],
-};
-
-const woman = {
-  species: 'human',
-  name: 'Leeloo Dallas',
-  gender: 'female',
-  legs: 2,
-  hands: 2,
-  saying: 'People hi!',
-  friends: [],
-};
-
-const man = {
-  species: 'human',
-  name: 'Korben Dallas',
-  gender: 'male',
-  legs: 2,
-  hands: 2,
-  saying: 'Hello there!',
-  friends: [],
-};
-
-// define cat-woman
-const catWoman = { ...woman };
-catWoman.name = 'Cat-woman';
-catWoman.saying = cat.saying;
+class Inhabitant {
+  constructor ({ species, name, gender, legs = 0, hands = 0, saying, friends = [] }) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.hands = hands;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  getFriendsList () {
+    if (this.friends.length > 0) {
+      return this.friends.map(friend => friend.name).join(', ');
+    }
+    return 'No friends!';
+  }
+
+  toString () {
+    return `${this.species}; ${this.name}; ${this.gender}; ${this.legs}; ${this.hands}; ${this.saying}; ${this.getFriendsList()}`;
+  }
+
+  addFriends(friends){
+    friends.forEach(friend => {
```

Comment:
Just for information:

1. `Array#push` supports multi-arrity - if you pass several items in it, it will add these multiple items into the array. So you can write `this.friends.push(...friends)`
2. You can use `Array#concat` - `this.friends = this.friends.concat(friends)`, or...
3. Spread - `this.friends = [...this.friends, ...friends]`

Reply:
this.friends = this.friends.concat(friends) - is so easy and so understandable... shame on me... ))))
I read about the spread operator before. Because madmaxWMFU suggested me to use it 7 days ago in a-tiny-JS-task  (https://github.com/kottans/frontend-2021-homeworks/pull/40#discussion_r515110274)
Of course, that way is the shorter syntax ... And of course, you are right in this case...
I just still don't recognize the relevance of using it... and I hope it will come with experiences ...